dev/ProjectReunion_BootstrapDLL/MddBootstrap.cpp

```diff
@@ -0,0 +1,377 @@
+                if (*pathWithoutFrameworkPath == L';')
+                {
+                    ++pathWithoutFrameworkPath;
+                }
+                (void)LOG_IF_WIN32_BOOL_FALSE(SetEnvironmentVariableW(L"PATH", pathWithoutFrameworkPath));
+            }
+            else
+            {
+                (void)LOG_HR_MSG(E_UNEXPECTED, "PATH doesn't start with %ls", frameworkPath);
+            }
+        }
+    }
+}
+
+
+CLSID FindDDLM(const PACKAGE_VERSION minVersion)
+{
+    // Find the best fit
+    bool foundAny{};
+    PACKAGE_VERSION bestFitVersion{};
+    CLSID bestFitClsid{};
+
+    // Look for windows.appExtension with name="com.microsoft.projectreunion.ddlm.<majorversion>.<architecture>"
+    WCHAR appExtensionName[100]{};
+    wsprintf(appExtensionName, L"com.microsoft.projectreunion.ddlm.%hu.%s", minVersion.Major, GetCurrentArchitectureAsString());
+
+    auto catalog{ winrt::Windows::ApplicationModel::AppExtensions::AppExtensionCatalog::Open(appExtensionName) };
+    auto appExtensions{ catalog.FindAllAsync().get() };
+    for (auto appExtension : appExtensions)
+    {
+        // Check the package identity against the package identity test qualifiers (if any)
+        if (!g_test_ddlmPackageNamePrefix.empty())
+        {
+            const auto packageId{ appExtension.Package().Id() };
+            std::wstring name{ packageId.Name().c_str() };
+            if ((name.rfind(g_test_ddlmPackageNamePrefix.c_str(), 0) != 0) ||
+                (CompareStringOrdinal(packageId.PublisherId().c_str(), -1, g_test_ddlmPackagePublisherId.c_str(), -1, TRUE) != CSTR_EQUAL))
+            {
+                // The package's Name prefix or PublisherId don't match the expected value. Skip it
+                continue;
+            }
+        }
+
+        // appExtension.Id == "ddlm-<major.minor.build.revision>-<architecture>"
+        const auto id{ appExtension.Id() };
+        PACKAGE_VERSION version{};
+        WCHAR architectureAsString[9 + 1]{};
+        const auto maxIdLength{ ARRAYSIZE(L"ddlm-12345.12345.12345.12345.abcdefghi") - 1 }; // -1 for length not counting null-terminator
+        if ((id.size() >= maxIdLength) ||
+            (swscanf_s(id.c_str(), L"ddlm-%hu.%hu.%hu.%hu-%9s", &version.Major, &version.Minor, &version.Build, &version.Revision, architectureAsString, static_cast<unsigned>(ARRAYSIZE(architectureAsString))) != 5))
+        {
+            (void)LOG_HR_MSG(ERROR_INVALID_DATA, "%ls", id.c_str());
+            continue;
+        }
+
+        // Does the version meet the minVersion criteria?
+        if (version.Version < minVersion.Version)
+        {
+            continue;
+        }
+
+        // Does the architecture match?
+        const auto architecture{ ParseArchitecture(architectureAsString) };
+        if (architecture != GetCurrentArchitecture())
+        {
+            continue;
+        }
+
+        // Do we yet to have a package under consideration?
```

Comment:
nit: "do we yet to have" - awkward phrasing

Reply:
Ack! I have no idea what guttural not-English I was muttering there...

Thanks